src/proof/proof_rule.h

```diff
@@ -23,9 +23,22 @@
 namespace cvc5 {
 
 /**
+ * \verbatim embed:rst:leading-asterisk
```

Comment:
Can you write the link in text of the html someone should look at to see the nice version of this header?

Reply:
It's in the initial comment of this PR: https://cvc5.github.io/docs-ci/docs-pr8303/proofs/proof_rules.html

Reply:
I'm saying that in the future, if someone looks at this header, we need a link to where they should look.

In particular, I'm concerned that after this PR the link from our IJCAR submission will point people to this document, and they will be unaware they need to go the html.